### PR TITLE
Replace console.time with performance api

### PR DESCRIPTION
**Pull request by eFrane**
Related issues: #75
Type of change:

[x] Code
[] Just documentation

If code was changed, did you...

[] ...write/update unit tests?
[] ...write/update documentation?

All existing tests still pass.

### CODE REVIEW

File: src/shared/Performance.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+export class Performance {
+  static hasPerformanceApi () {
+    const p = window.performance
+    return p && p.mark && p.measure && p.clearMarks && p.clearMeasures
+  }
+
+  static mark (tag) {
+    if (Performance.hasPerformanceApi()) {
```
Comment (salisdemos) [inline src/shared/Performance.js]: why `Performance` and not `this`?

File: src/shared/Performance.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      window.performance.mark(tag)
+    }
+  }
+
+  static measure (name, from, to) {
+    if (Performance.hasPerformanceApi()) {
+      window.performance.measure(name, from, to)
```
Comment (salisdemos) [inline src/shared/Performance.js]: same

File: src/shared/utils.js [hunk 3]
```diff
@@ -1,3 +1,14 @@
+
+/**
+ * Convenience wrapper to shorten the hasOwnProperty call
+ *
+ * @param {Object|Array{*}} object
+ * @param {String} property
+ */
+export function hasOwn (object, property) {
+  return Object.prototype.hasOwnProperty.call(object, property)
```
Comment (salisdemos) [inline src/shared/utils.js]: i think you forgot to move to `hasOwn` here

File: src/shared/Performance.js (hunk 1 above)
Comment (eFrane) [reply src/shared/Performance.js]: As the methods are all `static` they must be called statically. I just wrapped them into a class to make them more readable at their usage points.

File: src/shared/Performance.js (hunk 2 above)
Comment (eFrane) [reply src/shared/Performance.js]: S.a.

File: src/shared/utils.js (hunk 3 above)
Comment (eFrane) [reply src/shared/utils.js]: This is the `hasOwn` function definition? If I replace it here it'll just be a recursive no-op.

Comment (salisdemos) [reply src/shared/utils.js]: ups. Sorry.

**Comment by salisdemos** [comment]
also its not so much about console.time -> performanceApi and more about the refactoring to the hasOwn method. Looks all valid

**Comment by eFrane** [comment]
@salisdemos This needs another review since I merged the current master into it. Would you be so kind?

**Pull request merged**